examples/mixer_demo.rs

```diff
@@ -30,10 +32,13 @@ fn demo(filename: &Path) {
 
     let frequency = 44100;
     let format = AUDIO_S16LSB; // signed 16 bit samples, in little-endian byte order
-    let channels = 2; // Stereo
+    let channels = DEFAULT_CHANNELS; // Stereo
     let chunk_size = 1024;
     sdl2::mixer::open_audio(frequency, format, channels, chunk_size).unwrap();
-    sdl2::mixer::allocate_channels(0);
+
+    // Number of mixing channels available for sound effect `Chunk`s to play
+    // simultaneously.
+    sdl2::mixer::allocate_channels(4);
```

Comment:
Yes, exactly. If you `allocate_channel(0)` you cannot play sound effects.

See: https://www.libsdl.org/projects/SDL_mixer/docs/SDL_mixer_26.html

>Set the number of channels being mixed. This can be called multiple times, even with sounds playing. If numchans is less than the current number of channels, then the higher channels will be stopped, freed, and therefore not mixed any longer. It's probably not a good idea to change the size 1000 times a second though.
If any channels are deallocated, any callback set by Mix_ChannelFinished will be called when each channel is halted to be freed. Note: passing in zero WILL free all mixing channels, however music will still play.

The `rust-sdl2` docs for that function should probably be updated.